EDU 278: strongly typed errors update (#2007)
temporalio/documentation: docs-src/references/strongly-typed-errors/cause-unhandled-update.md

---
id: cause-unhandled-update
title: Cause Unhandled Update

> description: Explanation for Cause Unhandled Update error message, and how to fix it.

Reviewer: ```suggestion
description: Explanation for Cause Unhandled Update error message, and how to fix it.
```

From the description, this does not seem like something that needs to be fixed, it's just something that happens. Also doesn't seem like an error, but a warning/info? The Update is accepted and applied.

sidebar_label: Cause Unhandled Update
tags:
  - errors
  - strongly-typed
---

> `UnhandledUpdate` occurs when a Workflow Update is received by the server while a Workflow Task is being processed on a Worker producing a command that would cause the Workflow to transition to a closed state.

Reviewer 2: ```suggestion
`UnhandledUpdate` occurs when a [Workflow](/concepts/what-is-a-workflow) update is received by the Temporal Server while a Workflow Task being processed on a Worker produces a Command that would cause the Workflow to transition to a closed state.
```

> Temporal rejects the Workflow Task completion in order to guarantee that Updates are eventually seen by Workflow code and will rewind the Workflow so it can handle the pending Update.

Reviewer 2: ```suggestion
Temporal rejects the Workflow Task completion to guarantee that the update is eventually handled by Workflow code and rewinds the Workflow so it can handle the pending update.
```

> Temporal rejects the Workflow Task completion in order to guarantee that Updates are eventually seen by Workflow code and will rewind the Workflow so it can handle the pending Update.

Reviewer 2: ```suggestion
Temporal rejects the Workflow Task completion to guarantee that the update is eventually handled by Workflow code and rewinds the Workflow so it can handle the pending update.
```

> This error can happen when the [Workflow](/concepts/what-is-a-workflow) is receiving Updates at a high frequency.

Reviewer 3: Apart from needing to @djmagee ify these sentences I think my point comes across.

Reviewer 2: I've been verbed!

> This error can happen when the [Workflow](/concepts/what-is-a-workflow) is receiving Updates at a high frequency.

Reviewer 2: ```suggestion
This error can happen when the Workflow receives frequent updates.
```
Or possibly "...receives a high rate of updates."

> This error can happen when the [Workflow](/concepts/what-is-a-workflow) is receiving Updates at a high frequency.

Reviewer 2: ```suggestion
This error can happen when the Workflow receives frequent updates.
```
Or possibly "...receives a high rate of updates."

> This error can happen when the [Workflow](/concepts/what-is-a-workflow) is receiving Updates at a high frequency.

Reviewer 2: ```suggestion
This error can happen when the Workflow receives frequent updates.
```
Or possibly "...receives a high rate of updates."

---
id: cause-unhandled-update
title: Cause Unhandled Update
description: Explanation for Cause Unhandled Update error message, and how to fix it.
sidebar_label: Cause Unhandled Update
tags:
  - errors
  - strongly-typed
---

> `UnhandledUpdate` occurs when the [Workflow Execution](/concepts/what-is-a-workflow-execution) attempts to close itself before handling pending updates.

Reviewer 3: Need clarification what a "pending" update means here.
I'll come up with something more accurate here.

Reviewer 4: @bergundy any additional thoughts here? would like to close this it.

> This error can happen when the [Workflow](/concepts/what-is-a-workflow) is receiving a high number of [Signals](/application-development/features/#signals).

Reviewer: is there more context here? i don't understand why. is this specific to CAN?

> Drain the Signal Channel with the `ReceiveAsync` function to prevent or resolve this error.

Reviewer: sounds Go-specific, maybe say `In Go, ` and add a TODO comment to link to a new dev guide section on completing signal handling?

Reviewer 3: Draining the signals doesn't prevent the error.
Are you sure this is referring to signals? What is "update" in this context?

Author: Getting info from @alexshtin and others as I go along.

Reviewer 3: Not relevant, I'm resolving this conversation, this error is about workflow updates, not signals.

If this error persists, the Workflow may be on a different [Worker](/concepts/what-is-a-worker) than the one with the pending updates.

> Check your logs for failing [Workflow Tasks](/tasks/#workflow-task).

Reviewer 3: I can't make much sense out of this sentence.

Reviewer 3: I'm resolving and will offer alternative content.
